src/cargo/ops/cargo_rustc/context.rs

```diff
@@ -483,6 +483,16 @@ impl<'a, 'cfg> Context<'a, 'cfg> {
         // when changing feature sets each lib is separately cached.
         self.resolve.features_sorted(unit.pkg.package_id()).hash(&mut hasher);
 
+        if let Ok(dep_units) =  self.dep_targets(&unit) {
+            let mut dep_metadatas = dep_units.into_iter().map(|dep_unit| {
+                self.target_metadata(&dep_unit)
+            }).collect::<Vec<_>>();
+            dep_metadatas.sort();
+            for metadata in dep_metadatas {
+                metadata.hash(&mut hasher);
+            }
```

Comment:
is it cleaner to just do `dep_metadatas.hash(&mut hasher)` and rely on Vec's impl of Hash?

As a follow up, would it be nicer to use a sorted datastructure (like BTreeSet) instead of calling .sort() on a Vec? I'm certain it doesn't matter from a perf perspective, so whatever the team likes stylistically is fine to me.